### ISSUE TITLE
Cop idea: `string_literal.to_sym`

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#9353]
I was surprised to see that RuboCop doesn't pickup on this in a recent PR:

```ruby
# bad
'string-literal'.to_sym
# good
:'string-literal'
```

**Comment by koic** [comment]
I think this can be suggested to the Ruby style guide :-)

**Comment by bbatsov** [comment]
I guess this cop should also check for cases like `'string_literal'.to_sym` and `:'string_literal'`, as obviously those should have been `:string_literal`.

**Comment by deepj** [comment]
This cop would check if it starts with upper case letter or lower case as well. Because I don't think this is something what I would expect in hashes

```
~/utils.rb:13:11: W: [Correctable] Lint/SymbolConversion: Unnecessary symbol conversion; use UAE: instead.
          'UAE': 'United Arab Emirates',
          ^^^^^

~/utils.rb:16:11: W: [Correctable] Lint/SymbolConversion: Unnecessary symbol conversion; use Vietnam: instead.
          'Vietnam' 'Viet Nam',
          ^^^^^^^^^

```

The both cased wouldn't be symbols at all.

**Comment by marcandre** [comment]
@deepj Why not? keys may start with uppercase, can also be ruby keywords:
```ruby
h = {
  ABC: 42,
  class: :hello,
  end: 666,
} # => no problem
```

**Comment by dvandersluis** [comment]
Exactly what @marcandre said. `'UAE': 'United Arab Emirates'` maybe looks like the key will be a string, but it is already a symbol and the quotes are redundant.

```ruby
{ 'UAE': 'United Arab Emirates' }
=> {:UAE=>"United Arab Emirates"}
```

**Comment by deepj** [comment]
@dvandersluis I know it's valid. But I don't think it would be preferable default settings of Rubocop. It looks weird and unusual hash style (upper case letters in keys)

The part of hash does not come from Rails application with HashWithIndifferentAccess. Keys are searched by string dynamically. I don't write `states[:UAE]` or something like that. The cop it is useful. But not in case of hashes.

**Comment by dvandersluis** [comment]
The cop doesn't change `states['UAE']` to `states[:UAE]`, so you can still access your hash the same as you were previously. The only difference is that the extraneous quotes are removed from the symbol itself (again, your key is *not* a string, but a symbol, despite the quotes). I don't agree that hashes should not be processed, as I would much rather than `UAE: foo` than `'UAE': foo`.

**Comment by marcandre** [comment]
@deepj Maybe you meant to write `{'UAE' => 'United Arab Emirates'}`? IMO, user facing labels (like 'UAE') should be strings, not symbols.

Let's try and keep `HashWithIndifferentAccess` out of this, because I can't be partial about this (it's an abomination that was always a bad idea that is no longer needed with strong params)

**Comment by deepj** [comment]
I apologize, I'm just mixed several things together.
